src/main/java/org/folio/rest/service/action/ActionService.java

```diff
@@ -148,6 +148,9 @@ protected Feefineaction createFeeFineActionAndUpdateAccount(Account account, Mon
       .withDateAction(new Date());
 
     account.getPaymentStatus().setName(actionType);
+    account.setDateUpdated(new Date());
+    account.getMetadata().setUpdatedDate(new Date());
```

Comment:
Do you know which one of these fields is used on FE for "Date updated" column? Also, I though that `metadata` fields should be taken care of by RMB, but apparently they aren't?

Reply:
FE uses `metedata: updatedDate`. 
It will be little bit weird if  an account is updated and only `metedata: updatedDate` will be changed. In this case account  dateUpdated seems uncessary and duplicates "metedata: updatedDate".

Account is updated via PostgresClient, that's why RMB doesn't update `metadata` . Or RMB  should handle it too?

Reply:
Yes, it should. See `schema.json`:
```
      "tableName":"accounts",
      "fromModuleVersion":"15.0",
      "withMetadata":true
```

I don't think `Account.dateUpdated` was ever used. At least I could not find a single account instance with this property. We should not remove it from the schema though, as technically it would be a breaking change.

Reply:
According  [RMB  doc](https://github.com/folio-org/raml-module-builder#metadata) the `metadata` is updated by [triggers](https://github.com/folio-org/raml-module-builder/blob/master/domain-models-runtime/src/main/resources/templates/db_scripts/metadata.ftl). However only `created...` fields is hadled by the trigger.

Reply:
I wonder if this is a bug in RMB. In the doc it says that "RMB will set both update and create values to the same date/time and to the same user"

Reply:
I would like to retain these changes in PR. 

Removing `Account.setDateUpdated`  can be confusing in the future.
`Metadata..setUpdatedDate`  actually fixes the [issue](https://issues.folio.org/browse/MODFEE-150).  

As for RMB we can create ticket to create DB trigger on update action.

Reply:
Is this the only place in the code where we update an account? If it's not, we should update `Account.dateUpdated` in all those places too.

Reply:
I will go with your suggestion.

Reply:
I just tried debugging the whole thing. Looks like metadata is populated in an entity before it event gets to the implementation of a generated interface (*API class). When RMB converts JSON-body of the request into a POJO entity, it checks if entity has a `setMetadata()` method. And if it does, RMB uses this method to populate metadata.

Since in this case we did not receive `Account` in the request, its metadata is never updated. And for the same reason none of our `FeeFineActions` contain metadata.

RMD uses this method internally to populate metadata:
```
MetadataUtil.populateMetadata(Object entity, Map<String, String> okapiHeaders)
```
You can use it too to fix your problem. Just use headers you pass into the constructor of `ActionService` as second argument.

Reply:
@OleksandrVidinieiev  Thanks, nice find!